Project/Gem/Source/Vision/IdealVisionSystem.cpp

```diff
@@ -0,0 +1,305 @@
+        }
+        else
+        {
+            AZ_Error("IdealVisionSystem", false, "Failed to cook PhysX mesh");
+        }
+    }
+
+    void IdealVisionSystem::Deactivate()
+    {
+        ROS2SensorComponent::Deactivate();
+        m_detection3DPublisher.reset();
+        m_detection2DPublisher.reset();
+        m_detectionArrayPublisher.reset();
+    }
+
+    void IdealVisionSystem::Reflect(AZ::ReflectContext* context)
+    {
+        if (AZ::SerializeContext* serialize = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serialize->Class<IdealVisionSystem, ROS2SensorComponent>()->Version(1)->Field(
+                "ExcludeEntities", &IdealVisionSystem::m_excludeEntities);
+
+            if (AZ::EditContext* ec = serialize->GetEditContext())
+            {
+                ec->Class<IdealVisionSystem>("IdealVisionSystem", "IdealVisionSystem.")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "IdealVisionSystem")
+                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC("Game"))
+                    ->Attribute(AZ::Edit::Attributes::Category, "ROS2::Demo")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default, &IdealVisionSystem::m_excludeEntities, "ExcludeEntities", "ExcludeEntities");
+            }
+        }
+    }
+
+    void IdealVisionSystem::GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required)
+    {
+        required.push_back(AZ_CRC("ROS2CameraSensor"));
+    }
+
+    void IdealVisionSystem::FrequencyTick()
+    {
+        AZ::Transform sensorTransform;
+        AZ::TransformBus::EventResult(sensorTransform, GetEntityId(), &AZ::TransformBus::Events::GetWorldTM);
+        auto* sceneInterface = AZ::Interface<AzPhysics::SceneInterface>::Get();
+        AzPhysics::SceneHandle defaultSceneHandle = sceneInterface->GetSceneHandle(AzPhysics::DefaultPhysicsSceneName);
+        const AZ::Transform sensorTransformInv = sensorTransform.GetInverse();
+
+        geometry_msgs::msg::PoseArray poseArray;
+        vision_msgs::msg::Detection2DArray detection2DArray;
+        vision_msgs::msg::Detection3DArray detection3DArray;
+
+        std_msgs::msg::Header header;
+        header.frame_id = GetFrameID().c_str();
+        header.stamp = ROS2Interface::Get()->GetROSTimestamp();
+
+        poseArray.header = header;
+        detection3DArray.header = header;
+
+        // Overlap query with frustum
+        if (m_cookedMesh && sceneInterface && defaultSceneHandle != AzPhysics::InvalidSceneHandle)
+        {
+            AzPhysics::OverlapRequest request;
+            request.m_shapeConfiguration = m_cookedMesh;
+            request.m_pose = sensorTransform;
+            request.m_filterCallback = nullptr;
+            AzPhysics::SceneQueryHits results = sceneInterface->QueryScene(defaultSceneHandle, &request);
+            AZ_Printf("IdealVisionSystem", "Overlap with %d objects", results.m_hits.size());
```

Comment:
Looks like a debug line. This will clutter the O3DE console logs as it is printed very tick.